include/lbann/data_store/data_store_conduit.hpp

```diff
@@ -270,11 +346,14 @@ private :
   bool m_is_setup = false;
 
   /// set to true if data_store is preloaded
-  bool m_preload = false;
+  bool m_loading_is_complete = false;
+
+  /** @brief True, if we are in preload mode */
+  bool m_preloading = false;
 
   /// set to true if data_store is being explicitly loaded
   //VBE: please explain what this means!
```

Comment:
I'm unsure if you're referring to bot 349 and 352, but the answer is "yes." Loading is complete (for local cache mode) at the conclusion of exchange_local_caches(). This can occur after preloading, or after the first epoch if explicit loading. There is some redundancy here: m_preloading and m_explicitly_loading can not both be true, but both may be false. When m_loading_is_complete is true, both m_preloading and m_preloading should be false. I could have a single flag, "is_loading," but prefer to distinguish between the exact state of the code.
(unsure if that answers your question)